keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/IntentController.kt

```diff
@@ -73,7 +78,12 @@ class IntentController
       // TODO rz - Add updateStatus method to intentRepository
       throw NotImplementedError("soft-deleting intents is not currently supported")
     }
-    intentRepository.deleteIntent(id)
+    intentRepository.getIntent(id)
+      .takeIf { it != null }
+      ?.run {
+        applicationEventPublisher.publishEvent(AfterIntentDeleteEvent(this))
+        intentRepository.deleteIntent(id)
```

Comment:
Are these in the wrong order? Seems like the event should happen after the delete has been kicked off.

Reply:
"after" events are supposed to become "before" the actions they relate to. It sounds wrong because it is. :(

Reply:
I'm not even sure how this happened, but at least I was consistent.

Reply:
😂  you were very consistent